### PR TITLE
Added the blocked page hash view page [#154]

### PR DESCRIPTION
**Pull request by mcpierce**
* Added the blocked-page module.
 * Moved existing code to the new module.
 * Added a page for viewing the list of blocked page hashes.
 * Refactored all page-related code to a package named "page".
 * Added a new streaming session update model to the backend.

## Status
**READY**

## Migrations
NO

### CODE REVIEW

File: comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java [hunk 1]
```diff
@@ -459,7 +460,7 @@ public Page getPage(int index) {
   }
 
   /**
-   * Returns the number of pages associated with this comic.
+   * Returns the number of page associated with this comic.
```
Comment (BRUCELLA2) [inline comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java:463]: pages

File: comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java [hunk 2]
```diff
@@ -539,7 +540,7 @@ public int hashCode() {
   }
 
   public void removeDeletedPages(final boolean deletePages) {
-    log.debug("Remove deleted pages? {}", deletePages);
+    log.debug("Remove deleted page? {}", deletePages);
```
Comment (BRUCELLA2) [inline comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java:543]: pages

File: comixed-model/src/main/java/org/comixedproject/model/net/library/AddBlockedPageHashRequest.java [hunk 3]
```diff
@@ -24,8 +24,8 @@
 import lombok.NoArgsConstructor;
 
 /**
- * <code>AddBlockedPageHashRequest</code> represents the request body for a REST cal to blocked
- * pages by hash value.
+ * <code>AddBlockedPageHashRequest</code> represents the request body for a REST cal to blocked page
```
Comment (BRUCELLA2) [inline comixed-model/src/main/java/org/comixedproject/model/net/library/AddBlockedPageHashRequest.java:27]: REST call

File: comixed-model/src/main/java/org/comixedproject/model/session/SessionUpdateEvent.java [hunk 4]
```diff
@@ -16,38 +16,23 @@
...
-import com.fasterxml.jackson.annotation.JsonView;
 import java.util.Date;
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
-import org.comixedproject.views.View;
 
 /**
- * <code>AuditEvent</code> represents a single auditing event.
+ * <code>SessionUpdateEvent</code> represents a single auditing event.
```
Comment (BRUCELLA2) [inline comixed-model/src/main/java/org/comixedproject/model/session/SessionUpdateEvent.java:27]: represents a session event.

File: comixed-repositories/src/main/java/org/comixedproject/repositories/page/PageRepository.java [hunk 5]
```diff
@@ -39,11 +39,11 @@
   List<Page> getDuplicatePages();
 
   /**
-   * Marks all pages with the given hash as deleted.
+   * Marks all page with the given hash as deleted.
```
Comment (BRUCELLA2) [inline comixed-repositories/src/main/java/org/comixedproject/repositories/page/PageRepository.java:42]: pages

File: comixed-repositories/src/main/java/org/comixedproject/repositories/page/PageRepository.java [hunk 6]
```diff
@@ -39,11 +39,11 @@
...
    *
    * @param hash the hash
    * @param deleted the deleted state
-   * @return the number of pages marked
+   * @return the number of page marked
```
Comment (BRUCELLA2) [inline comixed-repositories/src/main/java/org/comixedproject/repositories/page/PageRepository.java:46]: pages

File: comixed-services/src/main/java/org/comixedproject/service/page/PageService.java [hunk 7]
```diff
@@ -163,7 +166,7 @@ public Page getForId(final long id) {
   }
 
   public List<Page> getAllPagesForComic(final long comicId) {
-    log.debug("Getting all pages for comic: id={}", comicId);
+    log.debug("Getting all page for comic: id={}", comicId);
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/page/PageService.java:169]: pages ?

File: comixed-services/src/main/java/org/comixedproject/service/page/PageService.java [hunk 8]
```diff
@@ -198,7 +201,7 @@ public int undeleteAllWithHash(final String hash) {
 
   @Transactional(isolation = Isolation.READ_UNCOMMITTED)
   public List<DuplicatePage> getDuplicatePages() {
-    log.debug("Getting pages from repository");
+    log.debug("Getting page from repository");
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/page/PageService.java:204]: pages ?

File: comixed-services/src/main/java/org/comixedproject/service/page/PageService.java [hunk 9]
```diff
@@ -223,7 +226,7 @@ public int undeleteAllWithHash(final String hash) {
   public void setDeletedState(final List<String> hashes, final boolean deleted) {
     for (int index = 0; index < hashes.size(); index++) {
       final String hash = hashes.get(index);
-      log.debug("Loading pages with hash: {}", hash);
+      log.debug("Loading page with hash: {}", hash);
```
Comment (BRUCELLA2) [inline comixed-services/src/main/java/org/comixedproject/service/page/PageService.java:229]: pages ?

File: comixed-tasks/src/main/java/org/comixedproject/task/model/ProcessComicWorkerTask.java [hunk 10]
```diff
@@ -91,7 +91,7 @@ public void startTask() throws WorkerTaskException {
       log.debug("Loading blocked page hashes");
       final List<String> blockedHashes = this.blockedPageHashService.getAllHashes();
 
-      log.debug("Checking for blocked pages");
+      log.debug("Checking for blocked page");
```
Comment (BRUCELLA2) [inline comixed-tasks/src/main/java/org/comixedproject/task/model/ProcessComicWorkerTask.java:94]: pages ?

File: comixed-tasks/src/main/java/org/comixedproject/task/model/ProcessComicWorkerTask.java [hunk 11]
```diff
@@ -100,7 +100,7 @@ public void startTask() throws WorkerTaskException {
       }
     }
 
-    logger.debug("Sorting pages");
+    logger.debug("Sorting page");
```
Comment (BRUCELLA2) [inline comixed-tasks/src/main/java/org/comixedproject/task/model/ProcessComicWorkerTask.java:103]: pages ?

File: comixed-web/src/app/blocked-page/services/blocked-page.service.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+} from '@app/blocked-page/blocked-page.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BlockedPageService {
+  constructor(private logger: LoggerService, private http: HttpClient) {}
+
+  /**
+   * Sets or clears the blacked state for a page's hsah.
```
Comment (BRUCELLA2) [inline comixed-web/src/app/blocked-page/services/blocked-page.service.ts:38]: blocked...
page's hash

**Review by BRUCELLA2 — COMMENTED** [review]
Few wording comments.
Coverage is under 80%. Couldn't some tests be added?

File: comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java (hunk 1 above)
Comment (mcpierce) [reply comixed-model/src/main/java/org/comixedproject/model/comic/Comic.java:463]: Ugh, I renamed a field from "pages" to "page" and IntelliJ went nuts and renamed things all over the place. :(

**Pull request closed without merging**